crates/derive/src/attr.rs

```diff
@@ -0,0 +1,149 @@
+//! Support for parsing and validating common attributes used by the different derive macros.
+
+use std::fmt;
+
+use syn::{Expr, ExprLit, Lit, Result};
+
+// Common trait all attributes should implement.
+pub(crate) trait Attribute
```

Comment:
Just to confirm, `Attribute` here can be any of `pliron`'s `Op`, `Attribute` or `Type` right?